src/rayleigh_quotient.jl

```diff
@@ -83,42 +116,55 @@ end
 function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_EIG)
-    b = prob.b
+    b = _get_b(prob) # fetch b matrix/vector if prob.b is span/vector
     C = prob.C
-    Kb = I - b * b' / dot(b, b)
-    nlargest = partialsortperm(eachrow(b), 1:size(b, 2); by=norm, rev=true)
     N = size(b, 1)
-    inds = map(!in(nlargest), 1:N)
-    J = I(N)[inds, :]
-    augC = J * Kb * C
+    Kb = I - b * b' / dot(b, b)
+    _, inds_keep = _inds_to_remove_and_keep(b, N)
+    augC = (Kb * C)[inds_keep, :] # instead of J * Kb * C where J = I(N)[inds_keep, :]
+    return _solve_homo_prob(augC, prob, alg)
+end
+
+function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_SPARSE)
+    b = _get_b(prob)
+    C = prob.C
+    inds_remove, inds_keep = _inds_to_remove_and_keep(b, length(b))
+    bk = first(b[inds_remove]) # only one element, no support for b Span
+    Ck = C[inds_remove, :]
+    augC = (C - (1/bk) * b * Ck)[inds_keep, :] # replace mult by J with slicing, see above
+    return _solve_homo_prob(augC, prob, alg)
+end
+
+function _inds_to_remove_and_keep(b, N)
+    inds_remove = partialsortperm(eachrow(b), 1:size(b, 2); by=norm, rev=true)
+    inds_keep = map(!in(inds_remove), 1:N)
+    return inds_remove, inds_keep
+end
+
+function _solve_homo_prob(augC, prob::ConstrainedRayleighQuotientProblem, alg)
+    b = _get_b(prob)
+    C = prob.C
     homo_prob = ConstrainedRayleighQuotientProblem(prob.Q, augC, zero(b))
     sol = solve(homo_prob, RQ_HOMO(alg.eps))
-    z = dot(b, C, sol) / dot(b, b)
-    return sol / z
+    t = dot(b, C, sol) / dot(b, b)
+    return sol / t
+end
+
+function _solve_homo_prob(augC, prob::ConstrainedRayleighQuotientProblem{Q,Cmat,<:Span} , alg) where {Q,Cmat}
+    b = _get_b(prob)[:, 1] # for b Span, normalization is not specified
+    homo_prob = ConstrainedRayleighQuotientProblem(prob.Q, augC, zero(b))
+    return solve(homo_prob, RQ_HOMO(alg.eps)) # returns a view, is that ok?
 end
 
 function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_HOMO)
     @assert iszero(prob.b) "b must be zero"
     C = prob.C
     P = I - C' * (factorize(Hermitian(C * C')) \ C)
     eig = eigen(Hermitian(P' * prob.Q.quadratic_form * P))
-    _select_vectors(eig.vectors, P, prob.b, alg.eps)
-end
-
-function _select_vectors(eigvecs, P, b::AbstractMatrix, eps=DEFAULT_SELECTVEC_EPS)
-    vecs = eachcol(eigvecs)
-    v = similar(first(vecs))
-    f = x -> any(>(eps) ∘ abs, mul!(v, P, x))
-    i1 = findfirst(f, vecs)
-    vecinds = [i1]
-    for n in 2:size(b, 2)
-        push!(vecinds, findnext(f, vecs, last(vecinds) + 1))
-    end
-    return reduce(hcat, vecs[vecinds])
+    _select_vectors(eig.vectors, P, alg.eps)
 end
 
-
-function _select_vectors(eigvecs, P, b::AbstractVector, eps=DEFAULT_SELECTVEC_EPS)
+function _select_vectors(eigvecs, P, eps=DEFAULT_SELECTVEC_EPS)
     vecs = eachcol(eigvecs)
     v = similar(first(vecs))
     f = x -> any(>(eps) ∘ abs, mul!(v, P, x))
```

Comment:
```suggestion
    f = x -> any(>(1/2), dot(x, P, x))
```
If P only has eigenvalues 0 and 1, then we could try this? This just checks if the result is closer to 0 or 1, which should be very robust.

Reply:
Yes, sounds smart! Could we do 

```julia
f = x -> dot(x, P, x) > 1/2
```